File: leet-code/intersection.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
  if (nums1.length === 0 || nums2.length === 0) {
    return []
  }
  const res = []
  const map = new Map()
  for (const item of nums1) {
    map.set(item, 1)
  }

  for (const item of nums2) {
    if (map.has(item)) {
      map.set(item, map.get(item) + 1)
    }
  }

  for (const [k, v] of map) {
    if (v > 1) {
      res.push(k)
    }
  }
  return res
}

File: leet-code/intersection.go
package main

import "fmt"

func intersection(nums1 []int, nums2 []int) []int {
	nMap := make(map[int]int, 0)
	res := make([]int, 0)
	if len(nums1) == 0 || len(nums2) == 0 {
		return res
	}

	for _, v := range nums1 {
		nMap[v] = 1
	}
	fmt.Printf("map=%#v", nMap)
	for _, v := range nums2 {
		if _, has := nMap[v]; has {
			nMap[v]++
		}
	}
	fmt.Printf("map=%#v", nMap)

	for key, val := range nMap {
		if val > 1 {
			res = append(res, key)
		}
	}
	return res
}
